addons/survey/static/src/views/survey_views.js

```diff
@@ -0,0 +1,82 @@
+/** @odoo-module **/
+
+import { registry } from '@web/core/registry';
+import { ListRenderer } from "@web/views/list/list_renderer";
+import { KanbanRenderer } from "@web/views/kanban/kanban_renderer";
+import { listView } from '@web/views/list/list_view';
+import { kanbanView } from '@web/views/kanban/kanban_view';
+import { useService } from "@web/core/utils/hooks";
+
+const { useComponent, useEffect, useRef } = owl;
+
+export function useSurveyLoadSampleHook(selector) {
+    const component = useComponent();
+    component.rootRef = useRef("root");
+    component.actionService = useService("action");
+    component.orm = useService('orm');
+    component.isLoadingSample = false;
+    useEffect(
+        (elems) => {
+            if (!elems || !elems.length) {
+                return;
+            }
+            const handler = (ev) => {
+                const surveyMethod = ev.currentTarget.closest('.o_survey_sample_container').getAttribute('action');
+                component.loadSample(surveyMethod);
+            }
+            for (const elem of elems) {
+                elem.addEventListener('click', handler);
+            }
+            return () => {
+                for (const elem of elems) {
+                    elem.removeEventListener('click', handler);
+                }
+            };
+        },
+        () => [component.rootRef.el && component.rootRef.el.querySelectorAll(selector)]
+    );
+    /**
+     * Load and show the sample survey related to the clicked element,
+     * when there is no survey to display.
+     * We currently have 3 different samples to load:
+     * - Sample Feedback Form
+     * - Sample Certification
+     * - Sample Live Presentation
+     */
+    component.loadSample = async function (method) {
```

Comment:
I wouldn't store a function on the component like that, as it's not meant to be called by anything else anyway